### PR TITLE
detect remote_addr if system is behind a proxy

### PR DESCRIPTION
**Pull request by dtdi**
found this useful to detect client ip when working behind a reverse proxy.

### CODE REVIEW

**Comment by bhadaway** [comment]
Hey Tobias,

Looks like the added bits for _stop-spammer-registrations-new.php_ were pulled from a very old version of Stop Spammers. This was before my time, but I would assume they were removed for a reason? Possibly it causes a conflict or memory issue?

After adding them back in have you tested thoroughly with the latest version of Stop Spammers and with PHP 7 as well? If you can't see any logical reason why they shouldn't be re-added, can you please do me a favor and un-format the code per the new semi-minified style?

Basically, just find `^\s*` and replace all with nothing for text editors that support Grep.

Thank you.
